pkg/completions/completions_test.go

```diff
@@ -0,0 +1,108 @@
+package completions_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ionos-cloud/ionosctl/v6/pkg/completions"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/json2table"
+	"github.com/stretchr/testify/assert"
+)
+
+type innerStruct struct {
+	Str     *string  `json:"string,omitempty"`
+	Integer *int32   `json:"int32,omitempty"`
+	Float   *float64 `json:"float64,omitempty"`
+}
+
+type outerStruct struct {
+	InnerStructs *[]innerStruct `json:"innerStructs,omitempty"`
+}
+
+var (
+	innerStructJsonPaths = map[string]string{
+		"int32":   "int32",
+		"string":  "string",
+		"float64": "float64",
+	}
+
+	testStr     = "test-string"
+	testInt32   = int32(123)
+	testFloat64 = float64(12.5)
+
+	testInnerStruct = innerStruct{
+		Str:     &testStr,
+		Integer: &testInt32,
+		Float:   &testFloat64,
+	}
+
+	testOuterStruct = outerStruct{
+		InnerStructs: &[]innerStruct{testInnerStruct, testInnerStruct},
+	}
+
+	testConvertedStruct []map[string]interface{}
+)
+
+func TestCompleter(t *testing.T) {
+	var err error
+
+	testConvertedStruct, err = json2table.ConvertJSONToTable("innerStructs", innerStructJsonPaths, testOuterStruct)
+	assert.NoError(t, err)
+
+	t.Run("initialize new completer", testNewCompleter)
+	t.Run("initialize new completer FAIL", testNewCompleterFail)
+	t.Run("add info to completer", testCompleterAddInfo)
+	t.Run("add info to completer WITH FORMATTING", testCompleterAddInfoWithFormat)
+	t.Run("add info to completer FAIL", testCompleterAddInfoFail)
+	t.Run("add MORE info to completer", testCompleterAddMoreInfo)
+	t.Run("add MORE info to completer FAIL", testCompleterAddMoreInfoFail)
+}
+
+func testNewCompleter(t *testing.T) {
+	completer := completions.NewCompleter(testConvertedStruct, "int32")
+	assert.NotEqual(t, completer, completions.Completer{})
+
+	printCompleter(completer)
+}
+
+func testNewCompleterFail(t *testing.T) {
+	completer := completions.NewCompleter(testConvertedStruct, "intz")
+	assert.Equal(t, completer, completions.Completer{})
+}
+
+func testCompleterAddInfo(t *testing.T) {
+	completer := completions.NewCompleter(testConvertedStruct, "int32").AddInfo("string")
+	assert.NotEqual(t, completer, completions.Completer{})
+
+	printCompleter(completer)
+}
+
+func testCompleterAddInfoFail(t *testing.T) {
+	completer := completions.NewCompleter(testConvertedStruct, "int32").AddInfo("stringz")
+	assert.Equal(t, completer, completions.Completer{})
+}
+
+func testCompleterAddInfoWithFormat(t *testing.T) {
+	completer := completions.NewCompleter(testConvertedStruct, "int32").AddInfo("string", "(%v)")
+	assert.NotEqual(t, completer, completions.Completer{})
+
+	printCompleter(completer)
```

Comment:
Can you also test the expected result, I for one frankly don't quite understand the intended output here when given the `additionalFormatting` parameter